inc/3rd-party/amazon-s3-and-cloudfront/inc/classes/class-imagify-as3cf.php

```diff
@@ -0,0 +1,280 @@
+<?php
+defined( 'ABSPATH' ) || die( 'Cheatin\' uh?' );
+
+/**
+ * Imagify WP Offload S3 class.
+ *
+ * @since  1.6.6
+ * @author Grégory Viguier
+ */
+class Imagify_AS3CF {
+
+	/**
+	 * Class version.
+	 *
+	 * @var string
+	 */
+	const VERSION = '1.0';
+
+	/**
+	 * Context used with get_imagify_attachment_class_name().
+	 * It matches the class name Imagify_AS3CF_Attachment.
+	 *
+	 * @var string
+	 */
+	const CONTEXT = 'AS3CF';
+
+	/**
+	 * An array containing the IDs (as keys) of attachments just being uploaded.
+	 *
+	 * @var array
+	 */
+	protected $uploads = array();
+
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var object
+	 */
+	protected static $_instance;
+
+	/**
+	 * Get the main Instance.
+	 *
+	 * Ensures only one instance of class is loaded or can be loaded.
+	 *
+	 * @since  1.6.6
+	 * @author Grégory Viguier
+	 *
+	 * @return object Main instance.
+	 */
+	public static function get_instance() {
+		if ( ! isset( self::$_instance ) ) {
+			self::$_instance = new self();
+		}
+
+		return self::$_instance;
+	}
+
+	/**
+	 * The class constructor.
+	 *
+	 * @since  1.6.6
+	 * @author Grégory Viguier
+	 */
+	protected function __construct() {
```

Comment:
Might be better to use a static method to call the hooks outside of the __construct() (see https://carlalexander.ca/designing-class-wordpress-hooks/)